operator/handler/handler.go

```diff
@@ -69,7 +78,77 @@ func (e *PodExtender) Handle(ctx context.Context, req admission.Request) admissi
 	if req.Operation == admv1.Delete {
 		// eliminates hanging kubectl apply -f command
 		// kube-apiserver server waits for response from operator on DELETE request
-		return admission.Allowed("Tailing Sidecar Operator does not block Pod deletion")
+		pod := &corev1.Pod{}
+		err := e.decoder.DecodeRaw(req.OldObject, pod)
+		if err != nil {
+			return admission.Allowed(fmt.Sprintf("Error ocurred (%v); %s", err, deletionMessage))
+		}
+
+		// Do not try to remove configMap in main namespace
+		if req.Namespace == e.ConfigMapNamespace {
+			return admission.Allowed(deletionMessage)
+		}
+
+		volumes := pod.Spec.Volumes
+
+		ret := true
+		for _, v := range volumes {
+			if v.Name == sidecarConfigurationName {
+				ret = false
+			}
+		}
+
+		if ret {
+			return admission.Allowed(deletionMessage)
+		}
+
+		key := types.NamespacedName{
+			Namespace: req.Namespace,
+			Name:      e.ConfigMapName,
+		}
+		configMap := corev1.ConfigMap{}
+		err = e.Client.Get(ctx, key, &configMap)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return admission.Allowed(deletionMessage)
+			}
+			return admission.Allowed(fmt.Sprintf("Error while getting configMap to clean up (%v); %s", err, deletionMessage))
+		}
+
+		podList := &corev1.PodList{}
+
+		listOptions := []client.ListOption{
+			client.InNamespace(req.Namespace),
+		}
+		err = e.Client.List(ctx, podList, listOptions...)
+		if err != nil {
+			return admission.Allowed(fmt.Sprintf("Error while getting list of pods (%v); %s", err, deletionMessage))
+		}
+
+		configMapUsed := false
+		for _, p := range podList.Items {
+			// skip current pod
+			if p.Name == pod.Name && p.Namespace == pod.Namespace {
+				continue
+			}
+
+			for _, volume := range p.Spec.Volumes {
+				if volume.ConfigMap != nil && volume.ConfigMap.Name == e.ConfigMapName {
+					configMapUsed = true
+					return admission.Allowed(deletionMessage)
+				}
+			}
+			if configMapUsed {
+				return admission.Allowed(deletionMessage)
+			}
```

Comment:
configMap may exist, but we may be in the state when it is not used by any pod

Reply:
we need to handle both scenarios:

- configMap exists and should be removed because nothing is using it
- configMap exists and we should leave it as it is being used by other objects

Reply:
Unless I'm not seeing something, we will never get the execute the conditional in line 141. We only set `configMapUsed` to true in 137, and then we always return immediately.

Reply:
Need to revalidate with current state, but according to this PR: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/56
the `kubelet` is in purpose